### DISCUSSION TITLE
Naming conventions

### DISCUSSION THREAD

**Discussion by samuelcastro** [UnlyEd/next-right-now#354] [General]
This is a very small think but I want to see your feedback. 

I think it'd be make more sense to rename `GraphCMSDataset` to just `Dataset`, I don't plan to use the dataset just for GraphCMS data and I think others will do the same.

Is there a reason behind this naming? Maybe you think it's better to create different dataset for different kind of data?

**Comment by Vadorequest** [comment]
I didn't expect people to merge several datasets coming from different sources into one. I'm not sure changing it would improve anything, this is usually an issue anyone customizing the code will run into.

**Reply by samuelcastro** [reply]
Yeah I get that, I'm curious about what would you do in this scenario.

I think keeping one single `Dataset` make more sense.

**Reply by Vadorequest** [reply]
I'd treat each data source differently. It'd also depend on whether they're shared, whether they should be made static, or whether they're needed in some particular pages.

Assuming it'd be similar from what's built-in in NRN, e.g Static data that are needed on all pages and must only fetched once, I'd preload them and load them in the `getStaticProps` and if the format is similar then merge it with the `dataset` returned by the function. Otherwise, I'd treat it completely differently, and the notion of a unique `dataset` wouldn't make as much sense and I would probably rename it to something more specialized. 

Really depends on many things, there is no answer for this. I'd do as I feel make the most sense, basically.

**Reply by samuelcastro** [reply]
Yeah, so I have data shared across all pages coming from more different data providers (GraphCMS and others), so I think I'll keep them in one single dataset that will be available statically, I don't think it makes sense to create different datasets, at some point I could have several datasets and it'd become hard to maintain.

But I do have data specific for individual pages, I think for those specific cases it does make sense to have different datasets.

I'd just rename `GraphCMSDataset` to `Dataset` though. :)

**Reply by samuelcastro** [reply]
Btw, I actually renamed it to `LayoutDataset`.

**Reply by Vadorequest** [reply]
Yeah, make sense too.
